editor/components/post-format/index.js

```diff
@@ -71,11 +71,16 @@ function PostFormat( { onUpdatePostFormat, postFormat = 'standard', supportedFor
 export default compose( [
 	withSelect( ( select ) => {
 		const { getEditedPostAttribute, getSuggestedPostFormat } = select( 'core/editor' );
-		const { getPostType } = select( 'core' );
-		const postType = getPostType( getEditedPostAttribute( 'type' ) );
+		const format = getEditedPostAttribute( 'format' );
+		const themeSupports = select( 'core' ).getThemeSupports();
+		let supportedFormats = get( themeSupports, 'formats', [] );
+		// Ensure current format is always in the set.
+		// The current format may not be a supported format.
+		supportedFormats.push( format );
+		supportedFormats = uniq( supportedFormats );
```

Comment:
Rather than the multi-step process for making `supportedFormats` unique, we can use the same pattern as we use in the PHP:

```js
const supportedFormats = union( [ format ], get( themeSupports, 'formats', [] ) );
```

Reply:
0d658a21